### PR TITLE
Eliminate dynamic shell commands

### PR DESCRIPTION
**Pull request by jze-spyro**
Each dynamic shell command is a potential bug, for example for paths containing white space, or any parameters containing any characters with special shell meaning (e.g. `*`, `"`, `'`, `!`, `$` etc.). Only static commands can be safely run as shell commands.

### CODE REVIEW

File: scargo/scargo_src/sc_docker.py [hunk 1]
```diff
@@ -52,15 +54,17 @@ def scargo_docker_run(
...
-    cmd = " ".join(
-        filter(
-            None,
-            ["docker-compose run", *docker_opts, f"{project_config_name}_dev", command],
-        )
-    )
+    cmd = [
+        "docker-compose",
+        "run",
+        *docker_opts,
```
Comment (fkuatspyro) [inline scargo/scargo_src/sc_docker.py:60]: What happens here if docker_opts is None?

Comment (jze-spyro) [reply scargo/scargo_src/sc_docker.py:60]: To answer your question: it would raise a `TypeError`. Same for an `int` or anything else that isn't iterable.

Comment (fkuatspyro) [reply scargo/scargo_src/sc_docker.py:60]: Maybe this was for something else I was thinking. Sequence should be right

File: scargo/scargo_src/sc_build.py [hunk 2]
```diff
@@ -47,13 +47,11 @@ def scargo_build(profile: str) -> None:
...
+            ["conan", "install", ".", "-if", build_dir],
             cwd=project_dir,
         )
         subprocess.check_call(
-            f"cmake -DCMAKE_BUILD_TYPE={profile} {project_dir}",
-            shell=True,
+            ["cmake", f"-DCMAKE_BUILD_TYPE={profile}", project_dir],
             cwd=build_dir,
         )
         subprocess.check_call("cmake --build . --parallel", shell=True, cwd=build_dir)
```
[the remark this answers is not in the document]
Comment (spyro-dgr) [reply scargo/scargo_src/sc_build.py:57]: No need to change since it is static command.

**Review by spyro-dgr — APPROVED** [review]

**Pull request merged**